sycl/test/host-interop-task/interop-task.cpp

```diff
@@ -0,0 +1,187 @@
+      Acc1[Id] = -1;
+      Acc2[Id] = -2;
+    });
+  });
+}
+
+// A test that uses OpenCL interop to copy data from buffer A to buffer B, by
+// getting cl_mem objects and calling the clEnqueueBufferCopy. Then run a SYCL
+// kernel that modifies the data in place for B, e.g. increment one, then copy
+// back to buffer A. Run it on a loop, to ensure the dependencies and the
+// reference counting of the objects is not leaked.
+void test1() {
+  static constexpr int COUNT = 4;
+  queue Q;
+  buffer<int, 1> Buffer1{BUFFER_SIZE};
+  buffer<int, 1> Buffer2{BUFFER_SIZE};
+
+  // init the buffer with a'priori invalid data
+  init<int, -1, -2>(Buffer1, Buffer2, Q);
+
+  // Repeat a couple of times
+  for (size_t Idx = 0; Idx < COUNT; ++Idx) {
+    copy(Buffer1, Buffer2, Q);
+    modify(Buffer2, Q);
+    copy(Buffer2, Buffer1, Q);
+  }
+
+  {
+    auto Acc = Buffer1.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "First buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the first buffer");
+    }
+  }
+  {
+    auto Acc = Buffer2.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "Second buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the second buffer");
+    }
+  }
+}
+
+// Same as above, but performing each command group on a separate SYCL queue
+// (on the same or different devices). This ensures the dependency tracking
+// works well but also there is no accidental side effects on other queues.
+void test2() {
+  static constexpr int COUNT = 4;
+  buffer<int, 1> Buffer1{BUFFER_SIZE};
+  buffer<int, 1> Buffer2{BUFFER_SIZE};
+
+  // init the buffer with a'priori invalid data
+  {
+    queue Q;
+    init<int, -1, -2>(Buffer1, Buffer2, Q);
+  }
+
+  // Repeat a couple of times
+  for (size_t Idx = 0; Idx < COUNT; ++Idx) {
+    queue Q;
+    copy(Buffer1, Buffer2, Q);
+    modify(Buffer2, Q);
+    copy(Buffer2, Buffer1, Q);
+  }
+
+  {
+    auto Acc = Buffer1.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "First buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the first buffer");
+    }
+  }
+  {
+    auto Acc = Buffer2.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "Second buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the second buffer");
+    }
+  }
+}
+
+// A test that does a clEnqueueWait inside the interop scope, for an event
+// captured outside the command group. The OpenCl event can be set after the
+// command group finishes. Must not deadlock according to implementation and
+// proposal
+void test3() {
+  // Want some large buffer for operation to take long
```

Comment:
I'd like to keep LIT tests as fast possible.
@vladimirlaz - FYI.

Reply:
The test runs for at most 8 seconds on my VM on CPU device with limitation to use only a single CPU core.
@bader is it OK for the test?

Reply:
There is no hard limit - it's just a recommendation to keep the development process efficient.
If it become a real problem, we can address this comment later.